File: API/API/Models/Page.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace API.Models
{
    public class Page<T>
    {
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public int NumberOfPages { get; set; }
        public int TotalNumberOfRecords { get; set; }
        public List<T> PageItems { get; set; }

    }
}

File: WebApp/src/app/models/page.ts
import { WorkItemComponent } from '../work-item/work-item.component';

export class Page {

    PageNumber : number;
    PageSize : number;
    NumberOfPages : number;
    TotalNumberOfRecords : number;
    PageItems : WorkItemComponent[];

    constructor()
    {

    }
    
}
